src/System.Private.CoreLib/shared/System/ThrowHelper.cs

```diff
@@ -399,6 +399,7 @@ internal static void IfNullAndNullsAreIllegalThenThrow<T>(object value, Exceptio
                 ThrowHelper.ThrowArgumentNullException(argName);
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         internal static void ThrowNotSupportedExceptionIfNonNumericType<T>()
```

Comment:
Maybe this name should be more explicit that it is a supported `S.R.Intrinsics.VectorXXX<T>` numeric type, so that it can cover future types (such as `Half`), if required

Reply:
Is a comment enough?

Reply:
Just calling it `ThrowForUnsupportedVectorBaseType` or something might be sufficient... Someone else may have a better idea

Reply:
Sounds good to me, will change.